docs/md/README.md

```diff
@@ -13,68 +13,166 @@ is a workload manager.
 ```bash
-$ make -jN
+$ make -j8
 ```
-where `N` is the number of concurrent jobs you wish to run.
-
-`make` will build each module in `worfklow-launcher`, run tests and generate
+where `8` can be replaced by any number that represents the concurrent
+jobs you wish to run.
+
+!!! info
+    If the version of your `make` is above GNU Make 4.0 (you could check
+    by running `make --version`), it's **highly recommended** to use
+    `--output-sync` along with `-j` so the standard outputs are sorted, i.e.
+    ```
+    $ make -j8 --output-sync
+    ```
+
+`make` will **build** each module in `workflow-launcher`, **run tests** and **generate**
 `Docker` images. All generated files go into a `derived` directory under the
 project root.
 
-You can invoke `make` on a module from the top level directory by
+You can also invoke `make` on a module from the top level directory by
 
 ```bash
 $ make [MODULE] TARGET={prebuild|build|check|images|clean|distclean}
 ```
 
-For most of the time, you would want to run:
+where currently available `MODULE`s are {api cloud_function docs helm ui}
+
+For most of the time, you would want to run something like:
+
+```bash
+$ make distclean
+```
+
+to clean up the built modules such as the docker images or the derived folder first,
+and then run:
+
+```bash
+$ make ui api TARGET=images -j8
+```
+
+to **only build** the WFL and its docker images without running tests.
+
+### Test
+
+If you only want to run tests on specific modules, you could run:
+
+```bash
+$ make [MODULE] TARGET=check
+```
+
+such as `make api TARGET=check` or `make cloud_function TARGET=check`.
+
+#### Clojure Test
+
+When it comes to clojure tests, sometimes it's useful to only run a subset
+of tests to save time and filter out noise. You can do this by directly
+invoke `clojure` cli from within the `api` directory, for example, `cd api` and:
 
 ```bash
-$ make ui api TARGET=images
+$ clojure -A:test integration --focus wfl.integration.v1-endpoint-test
+```
+
+In general, we implement Clojure tests under the `test/` root directory and use the
+[kaocha](https://cljdoc.org/d/lambdaisland/kaocha/1.0.632/doc/readme) test
+runner. Test suites use a `-test` namespace suffix. You can pass extra command
+line arguments to `kaocha`, such as the above `--focus` flag.
+You can see the full list of options with the following:
+
+```shell
+clojure -A:integration --help
+```
+
+At present, `wfl` api has two kinds of test, `unit` and `integration`. These can be run
+via the `deps.edn`, optionally specifying the kind:
+
+```shell
+clojure -A:test [unit|integration]
+```
+
+Note that the integration tests currently require a little more configuration
+before they can be run, namely, they require a `wfl` server running locally:
+
+```shell
+./ops/server.sh
 ```
-to only build the WFL and its docker images without running tests.
 
-See the output of `make help` for more information.
+### Deploy
+
+Currently, we mainly deploy WFL to gotc-dev and gotc-prod projects.
```

Comment:
We can be a bit more accurate here and say `broad-gotc-dev` and `broad-gotc-prod` (though we don't actually deploy to prod at all right now, I'm assuming this documentation is just forward-looking)